[ADD] Manufacturing: Production analysis report (#10155)
odoo/documentation: content/applications/inventory_and_mrp/manufacturing/workflows/production_analysis.rst

The *Production Analysis* report provides statistics about products manufactured using Odoo's
*Manufacturing* app. The report is useful when trying to understand production costs, manufacturing
durations, and other important statistics about manufactured products.

To open the Production Analysis report, navigate to :menuselection:`Manufacturing app --> Reporting
--> Production Analysis`.

.. important::
   The :guilabel:`Production Analysis` report is one of many reports available across the Odoo app
   suite. This documentation only covers the measures specific to the :guilabel:`Production
   Analysis` report, along with a few use case examples.

   For a full overview of the basic features available in most Odoo reports, see the documentation
   on :doc:`reporting essentials <../../../essentials/reporting>`.

Measures
========

*Measures* are the datasets that can be selected in the :guilabel:`Production Analysis` report. Each
dataset represents a specific statistic about |MOs| in the database. Choose a measure by clicking
the :guilabel:`Measures` :icon:`fa-caret-down` button, and selecting one of the options from the
drop-down menu:

> The options displayed in the :guilabel:`Measures` drop-down menu, and the order they appear in,

Reviewer: missing down-caret here?
```suggestion
The options displayed in the :guilabel:`Measures` :icon:`fa-caret-down` drop-down menu, and the order they appear in,
```

differ depending on the filters, groupings, and comparisons enabled in the :guilabel:`Search...`
bar. By default, the available measures appear as follows:

- :guilabel:`Average Employee Cost/Unit`: the average cost paid to employees to produce one unit of
  the product.
- :guilabel:`By-Products Total Cost`: the total value of all by-products created by manufacturing
  the product.

> - :guilabel:`Component Cost/Unit`: the average cost of the components required to produce
>   one unit of the product.

Reviewer: early line break
```suggestion
- :guilabel:`Component Cost/Unit`: the average cost of the components required to produce one unit
  of the product.
```

> - :guilabel:`Cost/Unit`: the average cost of producing one unit of the product,
>   including component, employee, operation, and subcontracting costs.

Reviewer: early line break
```suggestion
- :guilabel:`Cost/Unit`: the average cost of producing one unit of the product, including component,
  employee, operation, and subcontracting costs.
```

- :guilabel:`Duration of Operations/Unit`: the average total duration of operations required to
  produce one unit of the product.
- :guilabel:`Quantity Demanded`: the total number of units of the product included in |MOs|.
- :guilabel:`Quantity Produced`: the total number of units of the product that have actually been
  produced.
- :guilabel:`Total Component Cost`: the total amount spent on the product's components, across every
  |MO| for the product.
- :guilabel:`Total Cost`: the total amount spent manufacturing every unit of the product produced so
  far.
- :guilabel:`Total Duration of Operations`: the cumulative duration of every operation completed
  while manufacturing the product.
- :guilabel:`Total Employee Cost`: the cumulative amount paid to employees to manufacture the
  product.
- :guilabel:`Total Operation Cost`: the cumulative amount spent on operations required to produce
  the product.

> - :guilabel:`Total Operation Cost/Unit`: the average cost of the operations required to produce
>   one unit of the product.

Reviewer: early line break
```suggestion
- :guilabel:`Total Operation Cost/Unit`: the average cost of the operations required to produce one
  unit of the product.
```

- :guilabel:`Total Subcontracting Cost`: the cumulative amount paid to subcontractors to produce the
  product.
- :guilabel:`Total Subcontracting Cost/Unit`: the average cost of engaging a subcontractor to
  produce one unit of the product.
- :guilabel:`Yield Percentage (%)`: the total quantity of the product produced versus the total
  quantity demanded, represented as a percentage.
- :guilabel:`Count`: the total count of |MOs| created for the product.

> .. note::
>    Only one measure can be selected at a time.

Reviewer: This is true only in Graph view. On the Pivot table view, we can select and cross-reference multiple Measures as well as Group-bys (which is the more likely use case since we can also export into a spreadsheet)
![image](https://github.com/user-attachments/assets/8ed52728-7cfb-4bc6-8051-5aa62a5beec7)

Suggestion to reframe as a tip, something like this (I didn't include tag formatting)
```suggestion
.. tip::
   Only one measure can be selected at a time on the Graph view, however, multiple measures and varying group-by criteria (across the x and y axis) can be selected for deeper analysis on the Pivot table view.
```

> Use case: compare products

Reviewer: good example 👍

==========================

One of the best uses for the :guilabel:`Production Analysis` report is comparing statistics about
two or more products. This is accomplished by entering the products into the :guilabel:`Search...`
bar, then selecting the necessary measure, filter, and grouping, to see the desired data.

.. example::
   Toy manufacturer *Tommy's Toys* is trying to reduce their manufacturing operation costs. To
   accomplish this, they have decided to identify redundant products and cease manufacturing the

>    ones with higher operation costs.

Reviewer: nitpick, I think "operational" would be the descriptor for the subject "costs"
```suggestion
   ones with higher operational costs.
```

   the :guilabel:`Search...` bar, he enters the names of both products. Then, he opens the
   :guilabel:`Search...` bar drop-down menu, and clicks :guilabel:`Product` in the :guilabel:`Group
   By` section.

   Below the :guilabel:`Search...` bar, Mike clicks on the :guilabel:`Measures`
   :icon:`fa-caret-down` drop-down menu, and selects the :guilabel:`Total Operation Cost/Unit`
   option. Finally, he selects the :icon:`fa-bar-chart` :guilabel:`(bar chart)` graph type.

   With these options selected, the :guilabel:`Production Analysis` report shows a bar chart for the
   current year, with one bar for each product, signifying the average operation cost for one unit
   of the product.

   With this data, Mike is able to see that the average operation cost for the moon shoes is almost
   twice the cost of the pogo stick. Using this insight, Tommy's Toys decides to cease production of
   moon shoes, thus lowering their average cost of manufacturing operations.

   .. image:: production_analysis/use-case.png
      :align: center
      :alt: The bar chart comparing the operation costs of the pogo stick and moon shoes.

> Use case: compare time periods

Reviewer: Cool, didn't know graph view could be used for this.

      :guilabel:`(pie chart)` graph type, or the :icon:`oi-view-pivot` :guilabel:`(pivot)` view.

      A :guilabel:`Comparison` option can still be selected with the other view types enabled, but
      doing so does not change the way data is displayed on the report.

   Next, Adam selects the :guilabel:`Total Cost` option from the :guilabel:`Measures` drop-down
   menu. This option displays the total amount spent producing each product.

   In the :guilabel:`Search...` bar drop-down menu, he leaves the :guilabel:`2024` filter enabled in
   the :guilabel:`End Date` section, and enables the :guilabel:`Q2` filter as well. With both of
   these time periods selected, the pie chart shows data for the second quarter of 2024.

   Finally, Adam selects the :guilabel:`End Date: Previous Period` option in the
   :guilabel:`Comparison` section of the :guilabel:`Search...` bar. Doing so causes the pie chart to
   be split into an inner circle, and an outer ring.

   The outer ring shows data for the selected time period, quarter two of 2024. The inner circle
   shows data for the previous time period, quarter one of 2024.

   .. note::

>       If :guilabel:`End Date: Previous Year` is selected instead of :guilabel:`End Date:
>       Previous Period`, the inner circle shows data for the selected time period, one *year*
>       previous.

Reviewer: early line break
```suggestion
      If :guilabel:`End Date: Previous Year` is selected instead of :guilabel:`End Date: Previous
      Period`, the inner circle shows data for the selected time period, one *year* previous.
```

*Manufacturing* app. The report is useful when trying to understand production costs, manufacturing
durations, and other important statistics about manufactured products.

To open the production analysis report, navigate to :menuselection:`Manufacturing app --> Reporting
--> Production Analysis`.

Search bar options
==================

In the :guilabel:`Search...` bar at the top of the :guilabel:`Production Analysis` dashboard,
a variety of filters and groupings are available to refine the data shown in the report.

By default, the :guilabel:`Search...` bar includes the :guilabel:`End Date: [current year]` filter,
so that data is only displayed for |MOs| completed within the current calendar year. To remove this
filter, and show older data as well, click the :icon:`oi-close` :guilabel:`(Remove)` button on the
right side of the filter.

To only show the data for a specific product, type the product's name in the :guilabel:`Search...`
bar, and press enter. Multiple products can be entered into the search bar, so that only the data
for those products is shown.

In addition to typing in the :guilabel:`Search...` bar to refine the data shown, clicking the

> :icon:`fa-caret-down` :guilabel:`(toggle)` button on the right side of the bar opens a drop-down

Author: I like to call it the "toggle" button because it turns into an up arrow when you click on it.

The time periods to choose from are the three most recent months, each quarter of the current fiscal
year (:guilabel:`Q1`, :guilabel:`Q2`, :guilabel:`Q3`, :guilabel:`Q4`), and the three most recent
years.

.. note::
   Additional filter options can be selected by clicking the :guilabel:`Add Custom Filter` button.

Group By
--------

The :guilabel:`Group By` section of the drop-down menu is used to group manufacturing data into
specific categories. This makes each graph type show data based on that category, rather than
individual weeks.

.. example::
   With the bar chart graph type selected, clicking the :guilabel:`Product` option in the
   :guilabel:`Group By` section causes the chart to show one bar for each product, instead of one
   bar for each week during the selected time period.

Comparison
----------

The options in the :guilabel:`Comparison` section are used to compare production data from two
different time periods. There are two comparison options to choose from: :guilabel:`End Date:
Previous Period` and :guilabel:`End Date: Previous Year`.

To view data using one of the two comparisons, begin by selecting a time period in the

> :guilabel:`End Date` field of the :guilabel:`Search...` bar. Then, make sure to select either the

Author: I think it's fine to just say the search bar.

   *year* previous.

   If :guilabel:`End Date: Previous Period` is selected instead, the smaller circle shows data for
   second quarter of 2024, which is the same unit of time, but one *period* previous.

.. important::
   The :guilabel:`Comparison` section only appears in the :guilabel:`Search...` bar drop-down menu if
   one or more time periods is selected in the :guilabel:`Filters --> End Date` section. This is
   because a time period must be selected, or else there would be nothing to make a comparison to.

   Additionally, the comparison view is meant to be used with the :icon:`fa-pie-chart`
   :guilabel:`(pie chart)` graph type, or the :icon:`oi-view-pivot` :guilabel:`(pivot)` view.

.. image:: production_analysis/comparison.png
   :align: center
   :alt: The comparison view of the Production Analysis report.

Measures
========

*Measures* are the datasets that can be selected in the production analysis report. Each dataset
represents a specific statistic about |MOs| in the database. Choose a measure by clicking the
:guilabel:`Measures` :icon:`fa-caret-down` button, and selecting one of the following options from
the drop-down menu:

> 

Reviewer 2: I'm also noticing that some of these (in the list below) aren't showing up on the runbot I am using for this review... so, I'd strongly suggest a double-check on this

> 

Author: I've added a mention about how the options change based on different settings.

- :guilabel:`Average Component Cost/Unit`: the average cost of the components required to produce
  one unit of the product.
- :guilabel:`Average Employee Cost/Unit`: the average cost paid to employees to produce one unit of
  the product.
- :guilabel:`Average Subcontracting Cost/Unit`: the average cost of engaging a subcontractor to
  produce one unit of the product.
- :guilabel:`Average Total Cost/Unit`: the average cost of producing one unit of the product,
  including component, employee, operation, and subcontracting costs.
- :guilabel:`By-Products Total Cost`: the total value of all by-products created by manufacturing
  the product.
- :guilabel:`Duration of Operations/Unit`: the average total duration of operations required to
  produce one unit of the product.
- :guilabel:`Quantity Demanded`: the total number of units of the product included in |MOs|.
- :guilabel:`Quantity Produced`: the total number of units of the product that have actually been
  produced.
- :guilabel:`Total Component Cost`: the total amount spent on the product's components, across every

>   MO for the product.

Reviewer 2: ```suggestion
  |MO| for the product.
```

- :guilabel:`Total Cost`: the total amount spent manufacturing every unit of the product produced so
  far.
- :guilabel:`Total Duration of Operations`: the cumulative duration of every operation completed
  while manufacturing the product.
- :guilabel:`Total Employee Cost`: the cumulative amount paid to employees to manufacture the
  product.
- :guilabel:`Total Operation Cost`: the cumulative amount spent on operations required to produce
  the product.
- :guilabel:`Total Operation Cost/Unit`: the average cost of the operations required to produce
  one unit of the product.
- :guilabel:`Total Subcontracting Cost`: the cumulative amount paid to subcontractors to produce the
  product.
- :guilabel:`Yield Percentage (%)`: the total quantity of the product produced versus the total
  quantity demanded, represented as a percentage.
- :guilabel:`Count`: the total count of |MOs| created for the product.

.. note::
   Only one measure can be selected at a time.

Views
=====

> By default, the production analysis dashboard displays data using the *line chart* graph type of the

Reviewer 2: ```suggestion
By default, the :guilabel:`Production Analysis` dashboard displays data using the *line chart* graph type of the
```

Reviewer 2: might want to consider adding a graph-related icon here, as well

*graph view*.

> To select the bar chart or pie chart graph type instead, click the :icon:`fa-bar-chart`

Reviewer 2: ```suggestion
To select the bar chart or pie chart graph type, instead, click the :icon:`fa-bar-chart`
```

:guilabel:`(bar chart)`, or :icon:`fa-pie-chart` :guilabel:`(pie chart)` button below the
:guilabel:`Search...` bar. To return to the line chart, click the :icon:`fa-line-chart`

> :guilabel:`line chart` button.

Reviewer 2: ```suggestion
:guilabel:`(line chart)` button.
```

> To see production data displayed in a pivot table instead of a graph, click the

Reviewer 2: ```suggestion
To see production data displayed in a pivot table, instead of a graph, click the
```

> :icon:`oi-view-pivot` :guilabel:`(pivot view)`, button to the right of the :guilabel:`Search...`

Reviewer 2: ```suggestion
:icon:`oi-view-pivot` :guilabel:`(pivot view)` button, to the right of the :guilabel:`Search...`
```

bar. To return to graph view, click the :icon:`fa-area-chart` :guilabel:`(graph view)` button.

Bar chart

> ---------

Reviewer 2: might be cool to use anchors / ref links to these views, so the reader can quickly click-and-access them throughout the doc. Not required, just a suggestion to consider...

> The x-axis of the bar chart lists each week of the selected :guilabel:`End Date` time period. The

Reviewer 2: ```suggestion
The x-axis of the bar chart lists each week of the selected time period. The
```

> y-axis displays ascending numbers representing the count of whichever metric was selected on the

Reviewer 2: ```suggestion
y-axis displays ascending numbers, representing the count of whichever metric was selected on the
```

> :guilabel:`Measure` drop-down menu (ex. Average Total Cost/Unit). Thus, each bar represents the

Reviewer 2: ```suggestion
:guilabel:`Measures` drop-down menu (e.g. :guilabel:`Average Total Cost/Unit`). Thus, each bar represents the
```

total count during a single week of the selected time period.

> Each bar is divided into sections representing one of the products manufactured during that week. By

Reviewer 2: ```suggestion
Each bar is divided into sections, representing one of the products manufactured during that week. By
```

default, the sections are stacked on top of one another to form a single bar.

> To view the sections as individual bars, click the :icon:`fa-database` :guilabel:`(stacked)` button

Reviewer 2: ```suggestion
To view the sections as individual bars, click the :icon:`fa-database` :guilabel:`(stacked)` button,
```

to the right of the graph view buttons, to deactivate the stacking feature. Click the button again

> to return to the stacked view.

Reviewer 2: I'm wondering if 'stacked view' should be capitalized...or hit with italics... 🤔 it might just be good as-is. But, it's something to think about during your revisions. 👍

Mouse over a bar, or a section of a bar, to see a popover providing a few details about it:

> - The selected :guilabel:`Measure`.

Reviewer 2: ```suggestion
- The selected option from the :guilabel:`Measures` drop-down menu.
```

- The week and year.
- The product produced.

> - The total count of the selected :guilabel:`Measure`, during the week and year.

Reviewer 2: ```suggestion
- The total count of the selected :guilabel:`Measures` option, during the week and year.
```

Click on a bar, or a section of a bar, to open a page listing all of the |MOs| contained within it.

.. image:: production_analysis/bar-chart.png
   :align: center
   :alt: The bar chart graph type on the Production Analysis dashboard.

Line chart
----------

> The x-axis of the line chart lists each week of the selected :guilabel:`End Date` time period. The

Reviewer 2: ```suggestion
The x-axis of the line chart lists each week of the selected time period. The
```

> y-axis displays ascending numbers representing the count of whichever metric was selected on the
> :guilabel:`Measure` drop-down menu (ex. Total Component Cost). Thus, each point on the chart
> represents the total count during a single week of the selected time period.

Reviewer 2: ```suggestion
y-axis displays ascending numbers representing the count of whichever metric was selected from the
:guilabel:`Measures` drop-down menu (e.g. :guilabel:`Total Component Cost`). Thus, each point on the chart
represents the total count during a single week of the selected time period.
```

A uniquely colored line appears on the chart for each product manufactured during the selected

> :guilabel:`End Date` time period, connecting the points for each week. By default, the points on the

Reviewer 2: ```suggestion
time period, connecting the points for each week. By default, the points on the
```

chart, and the lines connecting them, are stacked on top of one another to show the total count of
all products for each week.

To view the points unstacked, click the :icon:`fa-database` :guilabel:`(stacked)` button to the
right of the graph view buttons, to deactivate the stacking feature. Click the button again to

> return to the stacked view.

Reviewer 2: same curious note re: 'Stacked' view as above ☝️

> Mouse over a week to see a popover providing a few details about the products produced during it:

Reviewer 2: ```suggestion
Mouse over a week to see a popover window providing a few details about the products produced during it:
```

> - The selected :guilabel:`Measure`.

Reviewer 2: ```suggestion
- The selected option from the :guilabel:`Measures` drop-down menu.
```

- A line for each product, showing:
  - The color used to represent the product.
  - The week and year.
  - The name of the product.

>   - The total count of the selected :guilabel:`Measure`, during the week and year.

Reviewer 2: ```suggestion
  - The total count of the selected :guilabel:`Measures` option, during the week and year.
```

.. image:: production_analysis/line-chart.png
   :align: center
   :alt: The line chart graph type on the Production Analysis dashboard.

Pie chart
---------

The pie chart displays one slice of pie for each product produced during each week of the selected

> :guilabel:`End Date` time period.

Reviewer 2: ```suggestion
time period.
```

.. example::

>    The month of February is selected in the :guilabel:`End Date` section of the
>    :guilabel:`Search...` bar. Units of a *chair* product and units of a *table* product were
>    manufactured during each of the four weeks of the month.

Reviewer 2: ```suggestion
   The month of February is selected from the :guilabel:`End Date` drop-down menu, located in the :guilabel:`Filters` column of the
   :guilabel:`Search...` bar mega menu of filters and grouping options. Units of a *chair* product, and units of a *table* product, were
   manufactured during each of the four weeks of the month.
```

   This means that the pie chart shows eight slices. Each slice represents one of the two products,
   and one of the four weeks in which it was produced.

> Mouse over a pie slice to see a popover providing a few details:

Reviewer 2: ```suggestion
Mouse over a pie slice to see a popover window providing a few details:
```

> - The selected :guilabel:`Measure`.

Reviewer 2: ```suggestion
- The selected option from the :guilabel:`Measures` drop-down menu.
```

- The week and year.
- The product produced.

> - The total count of the selected :guilabel:`Measure`, during the week and year.

Reviewer 2: ```suggestion
- The total count of the selected :guilabel:`Measures` option, during the week and year.
```

> - The percentage of the slice relative to the rest of the pie chart.

Reviewer 2: ```suggestion
- The percentage of the slice, relative to the rest of the pie chart.
```

Click on a slice to open a page listing all of the |MOs| contained within it.

.. image:: production_analysis/pie-chart.png
   :align: center
   :alt: The pie chart graph type on the Production Analysis dashboard.

Pivot view
----------

Pivot view displays production data in a pivot table. By default, each row of the table lists one
product, and the columns display the following data for each of those products:

- :guilabel:`Average Total Cost/Unit`
- :guilabel:`Average Component Cost/Unit`
- :guilabel:`Average Operation Cost/Unit`
- :guilabel:`Average Subcontracting Cost/Unit`
- :guilabel:`Quantity Produced`

To change the data categories displayed, click the :guilabel:`Measures` :icon:`fa-caret-down`

> button, and select or deselect categories from the popover window that appears.

Reviewer 2: ```suggestion
button, and select (or deselect) categories from the popover window that appears.
```

Click one of the column titles to sort the data from the lowest to the highest value. Click the
title again to sort from the highest to the lowest value.

> To make the rows of the pivot table display time periods instead of products, begin by clicking the

Reviewer 2: ```suggestion
To make the rows of the pivot table display time periods, instead of products, begin by clicking the
```

:icon:`fa-minus-square-o` :guilabel:`Total` button above the product names, which removes the
product rows, and changes the button to appear as :icon:`fa-plus-square` :guilabel:`Total`.

Click :icon:`fa-plus-square` :guilabel:`Total` to open a popover window showing options for the data
categories that can be displayed. Mouse over the :guilabel:`End Date` line to open a second popover
window showing time periods to choose from. Click one of the time periods, and the table updates to
show one unit of the period for each row.

.. image:: production_analysis/pivot-view.png
   :align: center
   :alt: The pivot view of the Production Analysis report.

Use case: compare products
==========================

> One of the best uses for the production analysis report is comparing statistics about two or more

Reviewer 2: ```suggestion
One of the best uses for the :guilabel:`Production Analysis` report is comparing statistics about two or more
```

products. This is accomplished by entering the products into the :guilabel:`Search...` bar, then
selecting the necessary measure, filter, and grouping, to see the desired data.

.. example::

>    Toy manufacturer *Tommy's Toys* is trying to reduce their manufacturing operation costs. To

Reviewer 2: ```suggestion
   Toy manufacturer, *Tommy's Toys*, is trying to reduce their manufacturing operation costs. To
```

>    accomplish this, they've decided to identify redundant products and cease manufacturing the ones

Reviewer 2: ```suggestion
   accomplish this, they have decided to identify redundant products, and cease manufacturing the ones
```

   with higher operation costs.

   Two of the toys that have been singled out for analysis are the *pogo stick* and *moon shoes*.

>    Tommy's Toys believes that these two toys are so similar that they can stop manufacturing one,

Reviewer 2: ```suggestion
   Tommy's Toys believes these two toys are so similar that they can stop manufacturing one,
```

   without significantly impacting their product offering.

>    To compare operation costs for the toys, business analyst Mike opens the

Reviewer 2: ```suggestion
   To compare operation costs for the toys, business analyst, Mike, opens the
```

>    :guilabel:`Manufacturing` app and navigates to the :guilabel:`Production Analysis` page. In the

Reviewer 2: ```suggestion
   :menuselection:`Manufacturing app`, and navigates to the :guilabel:`Production Analysis` page. In the
```

>    :guilabel:`Search...` bar, he enters the names of both products. Then, he opens the drop-down
>    menu, and clicks :guilabel:`Product` in the :guilabel:`Group By` section.

Reviewer 2: ```suggestion
   :guilabel:`Search...` bar, he enters the names of both products. Then, he opens the :guilabel:`Search...` bar drop-down
   menu of filter and grouping options, and clicks :guilabel:`Product` in the :guilabel:`Group By` section.
```

   Below the :guilabel:`Search...` bar, Mike clicks on the :guilabel:`Measures` :icon:`fa-caret-down`
   drop-down menu, and selects the :guilabel:`Total Operation Cost/Unit` option. Finally, he selects
   the :icon:`fa-bar-chart` :guilabel:`(bar chart)` graph type.

>    With these options selected, the production analysis report shows a bar chart for the current

Reviewer 2: ```suggestion
   With these options selected, the :guilabel:`Production Analysis` report shows a bar chart for the current
```

   year, with one bar for each product, signifying the average operation cost for one unit of the
   product.

>    With this data, Mike is able to see that the average operation cost for the moon shoes is almost

Reviewer 2: I know you've told me that certain formatting decisions have been made in the past with example admonitions like this that repeatedly mention a company or product, but I *do*, personally, think it looks better if, at the very least, the product name is capitalized... but, since you've said that you've been told to do otherwise, I guess this is just a pure suggestion to consider...
